src/Components/Endpoints/src/RazorComponentEndpointInvoker.cs

```diff
@@ -163,11 +163,22 @@ private async Task<RequestValidationState> ValidateRequestAsync(HttpContext cont
             // want to run the form handling logic against the error page.
             context.Features.Get<IExceptionHandlerFeature>() == null;
 
+
         if (processPost)
         {
-            var valid = false;
+            if (!context.Request.HasFormContentType)
+            {
+                context.Response.StatusCode = StatusCodes.Status400BadRequest;
+                if (context.RequestServices.GetService<IHostEnvironment>()?.IsDevelopment() == true)
```

Comment:
There's some logic in `EndpointHtmlRenderer.Streaming.cs` like this:

```cs
        var env = httpContext.RequestServices.GetRequiredService<IWebHostEnvironment>();
        var options = httpContext.RequestServices.GetRequiredService<IOptions<RazorComponentsServiceOptions>>();
        var showDetailedErrors = env.IsDevelopment() || options.Value.DetailedErrors;
```

I guess we should use a similar condition here, and maybe it could be factored out into a common place.

Reply:
This is why I hate partial classes, but good call

Reply:
I've factored out the code, and put it in a static method, I can't think of a good place to put that logic, so I'm going to just randomly leave it inside EndpointHtmlRenderer and use it in the invoker, if we go through this code again, we can do a clean-up pass.

Reply:
Wouldn't we also use `ShouldShowDetailedErrors` as the condition for whether to emit `The request has an incorrect Content-type.` to the response here, instead of `IsDevelopment` on line 171?

Reply:
Yes, I updated it in the wrong place